// ngx-dabd-2w1-core/esm2022/services/modal.service.mjs
import { inject, Injectable } from '@angular/core';
import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
import { from } from 'rxjs';
import { ModalConfirmComponent } from '../modal-confirm/modal-confirm.component';
import * as i0 from "@angular/core";
export class ModalService {
    ngModalService = inject(NgbModal);
    modalRef;
    showConfirm(settings) {
        this.modalRef = this.ngModalService.open(ModalConfirmComponent);
        this.modalRef.componentInstance.body = settings.body;
        this.modalRef.componentInstance.title = settings.title;
        this.modalRef.componentInstance.secondaryTitle = settings.secondaryTitle;
        this.modalRef.componentInstance.confirm = settings.confirmText;
        this.modalRef.componentInstance.cancel = settings.cancelText;
        return from(this.modalRef.result);
    }
    static ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.2.6", ngImport: i0, type: ModalService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
    static ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "18.2.6", ngImport: i0, type: ModalService, providedIn: 'root' });
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.2.6", ngImport: i0, type: ModalService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
